devito/types/grid.py

```diff
@@ -260,14 +259,15 @@ def _make_stepping_dim(self, time_dim, name=None):
             name = '%s_s' % time_dim.name
         return SteppingDimension(name=name, parent=time_dim)
 
+    @memoized_meth
     def _arg_defaults(self):
         """A map of default argument values defined by this Grid."""
         args = ReducerMap()
 
         for k, v in self.dimension_map.items():
             args.update(k._arg_defaults(_min=0, size=v.loc))
 
-        if configuration['mpi']:
+        if self.distributor.is_parallel:
```

Comment:
I didn't even remember I had added this property, lol

Reply:
:) Glad you did!